spec/requests/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe API::V1::UsersController, type: :request do
+  context 'when a user registers' do
+    it 'returns the user' do
+      post :create, params: create_user_params.merge!({ email: 'new_email@gmail.com' })
```

Comment:
We can omit `!` and `{}`
```suggestion
      post :create, params: create_user_params.merge(email: 'new_email@gmail.com' )
```

Reply:
I will remove the parentheses, though, for the `!` I had the same reaction as you when reading @ankitkalia1195  recommendation, but after some thought, I felt like "if `merge!` does not create a second variable, so it's probably more performant". 
As you raised the concern again, I decided to search for it :) 
So thanks to you, I've learned :) 
https://www.paperlesspost.com/blog/teams/different-methods-for-merging-ruby-hashes/

So even for super-small operations, I believe it's good to prefere merge! when it does not change anything else in the code :)

Reply:
Updated & pushed ;-)